core/src/main/java/feign/RequestTemplate.java

```diff
@@ -200,21 +200,29 @@ private static void putKV(String stringToParse, Map<String, Collection<String>>
     map.put(key, values);
   }
 
+  /** {@link #resolve(Map, Map)}, which assumes no parameter is encoded */
+  public RequestTemplate resolve(Map<String, ?> unencoded) {
+    return resolve(unencoded, Collections.<String, Boolean>emptyMap());
+  }
+
   /**
    * Resolves any template parameters in the requests path, query, or headers against the supplied
    * unencoded arguments. <br> <br><br><b>relationship to JAXRS 2.0</b><br> <br> This call is
    * similar to {@code javax.ws.rs.client.WebTarget.resolveTemplates(templateValues, true)} , except
    * that the template values apply to any part of the request, not just the URL
    */
-  public RequestTemplate resolve(Map<String, ?> unencoded) {
-    replaceQueryValues(unencoded);
+  public RequestTemplate resolve(Map<String, ?> unencoded, Map<String, Boolean> alreadyEncoded) {
```

Comment:
for this and below, package private is fine (as opposed to private). these are new methods, so we can hide them until someone asks. package private is better as it can be tested directly (plus the bytecode is smaller)